subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.java

```diff
@@ -403,8 +403,16 @@ private void compilerSettings(MavenProject project, BuildScriptBuilder builder)
         Plugin compilerPlugin = plugin("maven-compiler-plugin", project);
         if (compilerPlugin != null) {
             Xpp3Dom configuration = (Xpp3Dom) compilerPlugin.getConfiguration();
-            source = configuration.getChild("source").getValue();
-            target = configuration.getChild("target").getValue();
+            if (configuration != null) {
+                Xpp3Dom configuredSource = configuration.getChild("source");
+                Xpp3Dom configuredTarget = configuration.getChild("target");
+                if (configuredSource != null) {
+                    source = configuredSource.getValue();
+                }
+                if (configuredTarget != null) {
+                    target = configuredTarget.getValue();
```

Comment:
I suggest to check for the `null` or empty value of `configuredTarget.getValue()` to prevent overwriting null or empty-value on local variable `target` and ruining the default value, since the following config is still valid for maven:
```XML
<configuration>
    <source></source>
    <target></target>
</configuration>
```
(having the tags but no values)